core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -123,6 +129,39 @@
   private SqlFunctions() {
   }
 
+  /** SQL TO_BASE64(string) function. */
+  public static String toBase64(String string) {
+    return toBase64_(string.getBytes(UTF_8));
+  }
+
+  /** SQL TO_BASE64(string) function for binary string. */
+  public static String toBase64(ByteString string) {
+    return toBase64_(string.getBytes());
+  }
+
+  private static String toBase64_(byte[] bytes) {
+    String base64 = Base64.getEncoder().encodeToString(bytes);
+    StringBuilder str = new StringBuilder(base64.length() + base64.length() / 76);
+    if (base64.length() > 76) {
+      Splitter.fixedLength(76).split(base64).iterator().forEachRemaining(s -> {
```

Comment:
Can you please clarify why do you split the lines manually instead of using `java.util.Base64#getMimeEncoder()` ?
It looks like `Base64` can split the lines on its own just fine.

Reply:
> MIME
Uses the "The Base64 Alphabet" as specified in Table 1 of RFC 2045 for encoding and decoding operation. The encoded output must be represented in lines of no more than 76 characters each and uses a carriage return '\r' followed immediately by a linefeed '\n' as the line separator. No line separator is added to the end of the encoded output. All line separators or other characters not found in the base64 alphabet table are ignored in decoding operation.

I did it because I didn't find any evidence that MySql produces mime version of Base64 encoding and MySQL specifies that 
"A newline is added after each 76 characters of encoded output to divide long output into multiple lines." 
"Decoding recognizes and ignores newline, carriage return, tab, and space." 
I haved checked the binary result of TO_BASE64 in MySql , it seems that it not uses carriage return followed immediately by a linefeed as the line separator.